Dockerfile

```diff
@@ -11,12 +11,14 @@ VOLUME /var/lib/mysql
 COPY docker-entrypoint.d /docker-entrypoint.d
 COPY healthcheck.sh /opt/healthcheck.sh
 
-COPY docker-entrypoint.patch /usr/local/bin/docker-entrypoint.patch
+COPY docker-preinit-entrypoint.patch /usr/local/bin/docker-preinit-entrypoint.patch
+COPY docker-postinit-entrypoint.patch /usr/local/bin/docker-postinit-entrypoint.patch
 
 # Apply patch for running scripts placed in /docker-entrypoint.d/* by root
 RUN set -xe; \
-    sed -i '/\$(id -u)/ r /usr/local/bin/docker-entrypoint.patch' /usr/local/bin/docker-entrypoint.sh; \
-    rm -f /usr/local/bin/docker-entrypoint.patch
+    sed -i -e '/\$(id -u)/r /usr/local/bin/docker-preinit-entrypoint.patch' /usr/local/bin/docker-entrypoint.sh; \
```

Comment:
Please add comments to describe what each of the sed commands is doing, e.g. "inserts something at a particular place".